src/plugins/usage_collection/server/collector/collector_set.ts

```diff
@@ -83,14 +83,25 @@ export class CollectorSet {
       );
     }
 
-    const collectorTypesNotReady: string[] = [];
-    let allReady = true;
-    for (const collector of collectorSet.collectors.values()) {
-      if (!(await collector.isReady())) {
-        allReady = false;
-        collectorTypesNotReady.push(collector.type);
-      }
-    }
+    const collectorTypesNotReady = (
+      await Promise.all(
+        [...collectorSet.collectors.values()].map(async (collector) => {
+          if (!(await collector.isReady())) {
+            return collector.type;
+          }
+        })
+      )
+    ).filter((collectorType): collectorType is string => !!collectorType);
+    const allReady = collectorTypesNotReady.length === 0;
+
+    // const collectorTypesNotReady: string[] = [];
```

Comment:
Is there a reason to keep the "slow" code? If we're improving the performance, I don't see a need to keep what we're not using around.